clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -595,11 +598,19 @@ private long getLatencyInNS(long startTimeStamp) {
     return System.nanoTime() - startTimeStamp;
   }
 
+  private void checkPrerequisites() throws VeniceClientException {
+    if (!metadata.isReady()) {
+      throw new VeniceClientException("Store metadata is not ready");
```

Comment:
Let's be a little more verbose in these client facing logs. Include the store name and the fact that we will attempt to re-initialize it.